### ISSUE TITLE
[Notification Panel]: Dismiss all breaks focus flow between the notification panel and the global header

### ISSUE THREAD

**Issue by AlexanderMelox** [carbon-design-system/ibm-products#6010]
### Package

Carbon for IBM Products

### Description

When all notification are present you can tab through all of them and pressing escape puts focus back on the notification bell icon in the global header.

https://github.com/user-attachments/assets/0c940eac-15a9-4c82-936c-97222b86349d

When you dismiss all, pressing the escape key resets the tab. **Only happens when using keyboard navigation**

https://github.com/user-attachments/assets/d0e71ad5-ea65-4921-8cd4-a348657cc937

### Component(s) impacted

NotificationsPanel

### Browser

Chrome

### @carbon/ibm-products (previously @carbon/ibm-cloud-cognitive) version

v2.49.0

### Suggested Severity

Severity 4 = The problem is not visible to or noticeable to an average user. Affects minor functionality, no workaround needed.

### Product/offering

Library

### CodeSandbox or Stackblitz example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-patterns-notifications-notificationspanel--default&globals=viewport:basic

### Steps to reproduce the issue (if applicable)

1. Tab to the notifications panel
2. Open notifications panel with 'space'
3. Tab around a bit through notifications, the  shift tab back to dismiss all
4. Dismiss all with 'enter' key
5. press tab 2 times
6. press escape 
7. Loses focus

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate issues
